server/lib/db/transaction.ctrl.js

```diff
@@ -15,8 +15,48 @@ TransactionCtrl.handleError = (err, res, code, message) => {
   }))
 }
 
+TransactionCtrl.internal.createTransaction = (req, res) => {
```

Comment:
I think there's probably a cleaner way to do this rather than having nearly duplicate functions. We can refactor this later when I'm not so stretched for time.